### ISSUE TITLE
incorrectly flagging `python -m pip install ...`

### ISSUE THREAD

**Issue by bstpierre** [hadolint/hadolint#181]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

hadolint should not flag an error for a pinned pip package when run like `python -m pip install ...`.

### Actual behavior

I get an error when using `python -m pip install`, see below.

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
% docker run --rm -i hadolint/hadolint hadolint --version
Haskell Dockerfile Linter v1.6.0-0-gbd99295
% docker images | grep hadolint
hadolint/hadolint                                      latest              2b2411ca71ff        14 hours ago        59MB
```

**Dockerfile**

```Dockerfile
RUN pip install git+https://github.com/example/example@123abcdef
RUN pip install example==1.2.3
RUN python -m pip install example==1.2.3
RUN python -m pip install git+https://github.com/example/example@123abcdef
```

**Output**

```bash
% docker run --rm -i hadolint/hadolint < Dockertest
/dev/stdin:3 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`
/dev/stdin:4 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`
```

### See Also

Issue #67 is similar, and something like PR #99 may fix this. Adding `"python", "-m"` to the same location in [Rules.hs](https://github.com/hadolint/hadolint/pull/99/files#diff-7086fb4793c389a29a1ea9410da4098eR269) might fix the test case above, but note that in the original Dockerfile where I found this I'm using `RUN python3.6 -m pip install ...` and it doesn't seem ideal to have to enumerate all possible python versions in that list. I'm not really a haskell hacker so there may be a better solution.

**Comment by lorenzo** [comment]
Thanks for the report, we'll look into that

**Comment by jhoblitt** [comment]
I'm seeing the same bogus warning:
```
$ cat Dockerfile 
FROM alpine:3.7

ARG AWS_DEFAULT_REGION=us-east-1
ARG AWSCLI_VER=1.14.61

RUN apk add --no-cache --upgrade python3 bash && \
    pip3 install awscli=="${AWSCLI_VER}" --upgrade --no-cache-dir && \
    rm -rf /root/.cache
$ docker run --rm -i lukasmartinelli/hadolint < Dockerfile
/dev/stdin:6 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`
```

**Comment by zemanlx** [comment]
@bstpierre I do not see a reason to use `python -m pip` instead of `pip` (only if you are paid per character :wink: ), but correct me if I am wrong.

For `python3.6 -m pip install` mentioned later, can you use `pip3 install` instead?

**Comment by zemanlx** [comment]
@jhoblitt your issue has been solved a long time ago. All you need is to use the new image `hadolint/hadolint`.  

```bash
$ cat > Dockefile.181
FROM alpine:3.7

ARG AWS_DEFAULT_REGION=us-east-1
ARG AWSCLI_VER=1.14.61

RUN apk add --no-cache --upgrade python3 bash && \
    pip3 install awscli=="${AWSCLI_VER}" --upgrade --no-cache-dir && \
    rm -rf /root/.cache

$ docker run --rm -i hadolint/hadolint < Dockefile.181
/dev/stdin:6 DL3018 Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
```

**Comment by bstpierre** [comment]
> @bstpierre I do not see a reason to use python -m pip instead of pip

I only wrote it that way in the example because that is what I had minimized. I need to use `python3.6 -m pip` because this is running against ubuntu:16.04, which has python3.5 by default but I've also got a backported 3.6 and I need to pip install a couple of packages into the 3.6 environment. Running `pip3` would install into the 3.5 environment.

In that rules code if you partitioned the command at the point where `pip install` appears and discard the first part would that be a good general fix?

**Comment by jhoblitt** [comment]
Grrr.  The message I reported above is actually an `apk` warning.  The `pip` error is coming from `lukasmartinelli/hadolint`, which isn't present with `hadolint/hadolint`.  Please excuse the noise.

**Comment by zemanlx** [comment]
@bstpierre Ok, that sounds like a reasonable use case.
@jhoblitt No worries, it's Friday :tada:
